internal/services/synapse/synapse_spark_pool_resource.go

```diff
@@ -213,7 +214,12 @@ func resourceSynapseSparkPool() *pluginsdk.Resource {
 			"spark_version": {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
-				Default:  "3.4", // TODO: make this field Required in 4.0 since the default value changes over time
+				Default: func() string {
```

Comment:
Not push the changes in case for trigger the CI.
My understanding is keep `spark_version` same as before and add new schema for it but it is required, the code would like:
```go
if !features.FourPointOh() {
		resource.Schema["spark_version"] = &pluginsdk.Schema{
			Type:     pluginsdk.TypeString,
			Required: true,
			ValidateFunc: validation.StringInSlice([]string{
				"3.1",
				"3.2",
				"3.3",
				"3.4",
			}, false),
		}
	}
```